dataplane/linux/bpf_ep_mgr_test.go

```diff
@@ -126,27 +217,128 @@ var _ = Describe("BPF Endpoint Manager", func() {
 		},
 	}
 
+	hostEpNorm := proto.HostEndpoint{
+		Name: "uthost-eth0",
+		Tiers: []*proto.TierInfo{
+			&proto.TierInfo{
+				Name:            "mytier",
+				IngressPolicies: []string{"mypolicy"},
+				EgressPolicies:  []string{"mypolicy"},
+			},
+		},
+	}
+
 	It("does not have HEP in initial state", func() {
 		Expect(bpfEpMgr.hostIfaceToEpMap["eth0"]).NotTo(Equal(hostEp))
 	})
 
+	Context("with workload and host-* endpoints", func() {
+		JustBeforeEach(func() {
+			genIfaceUpdate("eth0", ifacemonitor.StateUp, 10)()
+			genWLUpdate("cali12345")()
+			genIfaceUpdate("cali12345", ifacemonitor.StateUp, 15)()
+			genHEPUpdate(allInterfaces, hostEpNorm)()
+		})
+
+		It("does not have host-* policy on the workload interface", func() {
+			Eventually(func() map[string]uint32 {
+				return dp.fds
```

Comment:
I'm afraid it is async because ensureProgramAttached and updatePolicyProgram are called from within goroutines.  So, as you say, we should have locks...